Treat status reports as non-critical (#2121)
github/codeql-action: CHANGELOG.md

## [UNRELEASED]

> - The upload-sarif action should not fail if it can't write to the telemetry api endpoint [#2121](https://github.com/github/codeql-action/pull/2121)

Reviewer: This is more than just the `upload-sarif` endpoint. I think this reads better.

```suggestion
- The CodeQL action no longer fails if it can't write to the telemetry api endpoint. [#2121](https://github.com/github/codeql-action/pull/2121)
```
